test/js/bun/shell/bunshell.test.ts

```diff
@@ -326,6 +326,20 @@ describe("bunshell", () => {
     expect(stdout.toString()).toEqual(`noice\n`);
   });
 
+  describe("tilde expansion", () => {
```

Comment:
can we have a test when `HOME` is not defined, empty, or invalid?

Reply:
also on windows will need to check the User env var

Reply:
Added test for empty or weird values of HOME, and in doing so found a ref counted related leak, so added test for that as well